### PR TITLE
Extinguisher QOL

### PR DESCRIPTION
**Pull request by XSlayer300**
## Changelog
:cl:
qol: Clicking yourself with the fire extinguisher while on HELP intent and with the safety off will extinguish yourself instead of bashing.
/:cl:

### CODE REVIEW

**Comment by ZahaSanko** [comment]
Does it also works with water cannon?

**Comment by XSlayer300** [comment]
> Does it also works with water cannon?

No, sadly. The flamer still acts as a weapon on its own right.

File: code/game/objects/items/tools/extinguisher.dm [hunk 1]
```diff
@@ -42,6 +42,12 @@
 	icon_state = "[sprite_name][!safety]"
 	to_chat(user, "The safety is [safety ? "on" : "off"].")
 
+/obj/item/tool/extinguisher/attack(mob/M, mob/user)
+	if(user.a_intent == INTENT_HELP && !safety) //If we're on help intent and going to spray people, don't bash them.
+		return FALSE
+	else
+		return ..()
```
Comment (TiviPlus) [inline code/game/objects/items/tools/extinguisher.dm:49]: No need for the else

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/tools/extinguisher.dm [hunk 2]
```diff
@@ -42,6 +42,10 @@
 	icon_state = "[sprite_name][!safety]"
 	to_chat(user, "The safety is [safety ? "on" : "off"].")
 
+/obj/item/tool/extinguisher/attack(mob/M, mob/user)
+	if(user.a_intent == INTENT_HELP && !safety) //If we're on help intent and going to spray people, don't bash them.
+		return ..()
```
Comment (TiviPlus) [inline code/game/objects/items/tools/extinguisher.dm:47]:
```suggestion
	if(user.a_intent == INTENT_HELP && !safety) //If we're on help intent and going to spray people, don't bash them.
		return FALSE
	return ..()
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**